### ISSUE TITLE
Clarify why PytorchModel and CnnResampleLoss subclasses exist

### ISSUE THREAD

**Issue by louisfh** [kitzeslab/opensoundscape#443]
I think we should clarify why the subclasses of PytorchModel and CnnResampleLoss (e.g. Resnet18Multiclass, InceptionV3 class) exist. As I understand this is for backwards compatibility? We could explain why these model classes are implemented in the docstrings/or somewhere else in the docs, and mention that `Resnet18Multiclass(classes)` is equivalent to `cnn.PytorchModel('resnet18',classes)...`

**Comment by sammlapp** [comment]
@louisfh there is one difference between `Resnet18Multiclass(classes)` and `PytorchModel('resnet18',classes)`:
- the former allows you to use separate training parameters (eg, learning rate) for the feature and classifier blocks
Do you think we should mark the Resnet18Multiclass and Resnet18Binary classes as deprecated? Then perhaps even remove them in 0.7.0? 
If this is more of a niche use, we could remove it from the codebase but put an example in a notebook incase we/someone else would like use that functionality. Alternitavely, we could leave it in - I just don't want to needlessly confuse the average user. 

On second thought, Since the Resnet18Binary and Resnet18Multiclass models now only appear in tutorials during "Advanced CNN training" specifically for the feature they are meant for, I think we can probably leave them without deprecating/removing. Avg user will never know they exist.
